src/npcmove.cpp

```diff
@@ -1655,6 +1656,12 @@ void npc::move_to( const tripoint &pt, bool no_bashing, std::set<tripoint> *nomo
             remove_effect( effect_bouldering );
         }
 
+        if( g->m.has_flag_ter_or_furn( TFLAG_NO_SIGHT, pos() ) ) {
+            add_effect( effect_no_sight, 1_turns, num_bp, true );
+        } else if( has_effect( effect_no_sight ) ) {
+            remove_effect( effect_no_sight );
+        }
```

Comment:
I haven't tested, but this doesn't seem right.  It looks like you're adding a 1-turn effect on moving to a location with NO_SIGHT, but won't that wear off after 1 turn if the NPC remains there?  An effect doesn't seem like the right approach for this feature.  Also, it would be unfortunate to need npc-specific code to make it work (as opposed to code for all creatures or at least all `player` objects (which currently includes `npc`).

Reply:
Also it's important to note that in this state it only works for player, monster on the tile can still see you even if you are two tiles away so I definitly missed something.
But it might just be that I did not understand correctly how monster sight works as the in_pit effect also only affect player's sight but not monster's.